src/section/help.rs

```diff
@@ -1,10 +1,76 @@
+use crate::{section, Report, Result, Section};
 use ansi_term::Color::*;
 use std::fmt::{self, Display};
 
 /// A helper trait for attaching help text to errors to be displayed after the chain of errors
+///
+/// `color_eyre` provides two types of help text that can be attached to error reports, custom
+/// sections and pre-configured sections. Custom sections are added via the `section` and
+/// `with_section` methods, and give maximum control over formatting. For more details check out
+/// the docs for [`Section`].
+///
+/// The pre-configured sections are provided via `suggestion`, `warning`, and `note`. These
+/// sections are displayed after all other sections with no extra newlines between subsequent Help
+/// sections. They consist only of a header portion and are prepended with a colored string
+/// indicating the kind of section, e.g. `Note: This might have failed due to ..."
+///
+/// [`Section`]: struct.Section.html
 pub trait Help<T>: private::Sealed {
-    /// Add a note to an error, to be displayed after the chain of errors.
+    /// Add a section to an error report, to be displayed after the chain of errors.
+    ///
+    /// Sections are displayed in the order they are added to the error report. They are displayed
+    /// immediately after the `Error:` section and before the `SpanTrace` and `Backtrace` sections.
+    /// They consist of a header and an optional body. The body of the section is indented by
+    /// default.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,should_panic
+    /// use color_eyre::{Report, Help};
+    /// use eyre::eyre;
+    ///
+    /// Err(eyre!("command failed"))
+    ///     .section("Please report bugs to https://real.url/bugs")?;
+    /// # Ok::<_, Report>(())
+    /// ```
+    fn section<C>(self, section: C) -> Result<T>
+    where
+        C: Into<Section>;
+
+    /// Add a section to an error report, to be displayed after the chain of errors, which is
+    /// lazily evaluated only in the case of an error
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use color_eyre::{Report, Help, SectionExt};
+    /// use eyre::eyre;
+    ///
+    /// let output = std::process::Command::new("ls")
+    ///     .output()?;
+    ///
+    /// let output = if !output.status.success() {
+    ///     let stderr = String::from_utf8_lossy(&output.stderr);
+    ///     Err(eyre!("cmd exited with non-zero status code"))
+    ///         .with_section(move || {
+    ///             "Stderr:"
+    ///                 .skip_if(|| stderr.is_empty())
+    ///                 .body(stderr.trim().to_string())
+    ///         })?
+    /// } else {
+    ///     String::from_utf8_lossy(&output.stdout)
+    /// };
+    ///
+    /// println!("{}", output);
+    /// # Ok::<_, Report>(())
+    /// ```
+    fn with_section<C, F>(self, section: F) -> Result<T>
+    where
+        C: Into<Section>,
+        F: FnOnce() -> C;
+
+    /// Add a note to an error report, to be displayed after the chain of errors.
```

Comment:
nit, consistency:

```suggestion
    /// Add a Note to an error report, to be displayed after the chain of errors.
```

Is the Note lazily evaluated?

Reply:
in the `note` fn any expression will be evaluated eagerly, kinda like the difference between `unwrap_or` and `unwrap_or_else` in std. However the type that is returned is just boxed internally, we don't do any String allocation or anything, we just hold onto the display type until its time to print the entire thing and we format its output into the final buffer. So in that sense it is lazy. So I guess to answer this, it depends on what you mean.